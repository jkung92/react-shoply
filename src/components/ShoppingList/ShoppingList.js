import React, { Component } from 'react';
import ItemContainer from '../../containers/ItemContainer';
//import './ShoppingList.css';

class ShoppingList extends Component {
  render() {
    return (
      <div className="ShoppingList">
        {this.props.items.map(item => (
          <ItemContainer
            id={item.id}
            price={item.price}
            name={item.name}
            image={item.image_url}
          />
        ))}
      </div>
    );
  }
}

ShoppingList.propTypes = {};

ShoppingList.defaultProps = {};

export default ShoppingList;
