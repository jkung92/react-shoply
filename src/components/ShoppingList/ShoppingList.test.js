import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ShoppingList from './ShoppingList';

it('renders without crashing', function() {
  let items = [
    {
      id: 1,
      name: 'tv',
      price: 219.99,
      image_url:
        'https://images.samsung.com/is/image/samsung/latin_en-hd-j4300ah-un32j4300ahxpa-001-front-indigo-blue'
    }
  ];
  shallow(<ShoppingList items={items} />);
});

it('matches snapshot', function() {
  let items = [
    {
      id: 1,
      name: 'tv',
      price: 219.99,
      image_url:
        'https://images.samsung.com/is/image/samsung/latin_en-hd-j4300ah-un32j4300ahxpa-001-front-indigo-blue'
    }
  ];
  const wrapper = shallow(<ShoppingList items={items} />);
  const serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});
