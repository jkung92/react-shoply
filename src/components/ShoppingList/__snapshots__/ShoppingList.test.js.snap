// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="ShoppingList"
>
  <Connect(ItemContainer)
    id={1}
    image="https://images.samsung.com/is/image/samsung/latin_en-hd-j4300ah-un32j4300ahxpa-001-front-indigo-blue"
    name="tv"
    price={219.99}
  />
</div>
`;
